PLEP-0007.rst

```diff
@@ -0,0 +1,364 @@
+|                 | | Any tools that go into defining a plasma or its           |
+|                 |   environment (e.g. a field solver) should be included in   |
+|                 |   a sub-package within ``plasmapy.plasma``.                 |
++-----------------+-------------------------------------------------------------+
+
+.. _simulation:
+
++-----------------+-------------------------------------------------------------+
+| **simulation**                                                                |
++=================+=============================================================+
+| directory:      | ``./plasmapy/simulation/``                                  |
++-----------------+-------------------------------------------------------------+
+| access:         | ``plasmapy.simulation``                                     |
++-----------------+-------------------------------------------------------------+
+| PLEP:           |                                                             |
++-----------------+-------------------------------------------------------------+
+| scope:          | | A sub-package focused on interfacing with simulations     |
+|                 |   and/or running simulations.                               |
+|                 | |                                                           |
+|                 | | If a new feature falls under the scope of the             |
+|                 |   ``analysis`` and/or ``diagnostics`` sub-packages, then    |
+|                 |   the new feature should be included in one of those        |
+|                 |   sub-packages.  For example, a synthetic diagnostic        |
+|                 |   should be included in the ``plasmapy.diagnostics``        |
+|                 |   sub-package.                                              |
++-----------------+-------------------------------------------------------------+
+
+.. _tests:
+
++-----------------+-------------------------------------------------------------+
+| **tests**                                                                     |
++=================+=============================================================+
+| directory:      | ``./plasmapy/tests/``                                       |
++-----------------+-------------------------------------------------------------+
+| access:         | ``plasmapy.tests``                                          |
++-----------------+-------------------------------------------------------------+
+| PLEP:           |                                                             |
++-----------------+-------------------------------------------------------------+
+| scope:          | | A collection of tests for top-level modules (i.e.         |
+|                 |   functions and classes defined in top-level ``.py``        |
+|                 |   files).                                                   |
```

Comment:
Surely we also want to test functions and classes defined in subfolders of these modules?

Reply:
I thought folders of .py files were subpackages, and if that's the case, the current wording makes sense - there's no other place to put tests for .py files in the main plasmapy directory.

Reply:
In the python lingo, `.py` files are modules and folders are sub-packages.

@SolarDrew Yes, sub-package tests should still be done within their respective directory.

Reply:
Think of this as a place for testing the interplay of various sub-packages. Will using dispersion capability in simulation break anything? Will using formulary inside analysis break anything?